test/unistd/access.c

```diff
@@ -65,6 +66,32 @@ int main() {
   printf("F_OK(%s): %d\n", "renamedfile", faccessat(AT_FDCWD, "renamedfile", F_OK, 0));
   printf("errno: %d\n", errno);
 
+
+  EM_ASM(
+    var fchmodstream = FS.open("fchmodtest", "r");
+#if WASMFS
+    FS.fchmod(fchmodstream, 0o0000);
+#else
+    FS.fchmod(fchmodstream.fd, 0o000);
+#endif
+  );
+
+#if !NODE*FS
+  EM_ASM(
+    FS.symlink('forbidden', 'symlinkfile');
+    FS.lchmod('symlinkfile', 0o777);
+  );
+  struct stat fileStats;
+  struct stat symlinkStats;
+
+  lstat("symlinkfile", &symlinkStats);
+  assert(symlinkStats.st_mode & (S_IRWXU | S_IRWXG | S_IRWXO));
```

Comment:
Can the `S_*` stuff be identical to the mode that is set above on line 82? That is, use `0o777` in both places, or use `S_*` in both?

Reply:
This makes sense. In general, would it be better to just use octal notation or the macros? I remember I made a similar mistake on some of my previous tests.

Reply:
Hmm, I don't have much of a preference myself. Perhaps the macros are slightly easier to read?